src/main/java/com/arpnetworking/clusteraggregator/aggregation/AggMessageExtractor.java

```diff
@@ -52,6 +56,20 @@ public String entityId(final Object message) {
                     .append(metricData.getMetric())
                     .append("||")
                     .append(metricData.getPeriod());
+            final ArrayList<Messages.DimensionEntry> sortedDimensions = new ArrayList<>(metricData.getDimensionsList());
+            sortedDimensions.sort(Comparator.comparing(Messages.DimensionEntry::getKey));
```

Comment:
No probs, I'll add de-duping. Any thoughts on memoization of the operation for other parts of the code to re-use?